packages/wp-now/src/wp-now.ts

```diff
@@ -0,0 +1,200 @@
+import fs from 'fs-extra';
+import { NodePHP } from '@php-wasm/node'
+import { SupportedPHPVersion } from '@php-wasm/universal'
+import path from 'path'
+import { SQLITE_FILENAME, SQLITE_PATH, WORDPRESS_ZIPS_PATH, WP_NOW_HIDDEN_FOLDER } from './constants'
+import { downloadSqlite, downloadWordPress } from './download'
+import { portFinder } from './port-finder'
+
+export interface WPNowOptions {
+  phpVersion?: SupportedPHPVersion
+  documentRoot?: string
+  absoluteUrl?: string
+  mode?: 'plugin' | 'theme' | 'core' | 'index'
+  projectPath?: string
+}
+
+const DEFAULT_OPTIONS: WPNowOptions = {
+  phpVersion: '8.0',
+  documentRoot: '/var/www/html',
+  mode: 'index',
+}
+async function getAbsoluteURL() {
+  const port = await portFinder.getOpenPort()
+  return `http://127.0.0.1:${port}`
+}
+
+function seemsLikeAPHPFile(path) {
```

Comment:
There's a lot of overlap with the service worker on the web, let's put it in some kind of `@wp-playground/common` package eventually.

Side note – it would be lovely to migrate to the `@wordpress` org and have packages like `@wordpress/playground-common` etc